### DISCUSSION TITLE
Sort Filter Run Prefix: Flag seems does not work with variables

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5941] [Q&A] [answered]
1. goto https://tiddlywiki.com/prerelease/
2. create three tiddlers with below titles all tagged with xtx as below
one: first
two: second
three: third

3. In another tiddler enter the below code

```
\define ff() :sort:$(sortType)$:$(sortNegate)$[split<delimiter>!is[blank]trim[]nth<pos>]

\define test2()

<$vars pos=2 sortType=string delimiter=":" sortNegate="reverse"> 
<$list filter="[tag[xtx]] $(ff)$ ">

</$list>
</$vars>
\end

<<test2>>
```

It does not work with reverse flag when the revers is sent as a variable!

But if you literally use the reverse, it works so the below code works correctly

```
\define ff() :sort:$(sortType)$:reverse[split<delimiter>!is[blank]trim[]nth<pos>]

\define test2()

<$vars pos=2 sortType=string delimiter=":" sortNegate="reverse"> 
<$list filter="[tag[xtx]] $(ff)$ ">

</$list>
</$vars>
\end

<<test2>>

```

Is this a bug in `:sort:type:flag[]` or something else is wrong?

**Comment by saqimtiaz** [comment]
Before `<<test2>>` add:

`<$log test2=<<test2>>/>`

Then check the browser developer console. I suspect you will see the issue is with the variable substitution in the nested macro.

**Reply by kookma** [reply]
Hi Saq
![327_chrome](https://user-images.githubusercontent.com/830394/128588002-58dcaf68-1d8e-4645-9c17-701595476ede.png)

This is what I get! both position for sortType and sortNegate are empty!

To reproduce I have attached a zip of all tiddlers

[tiddlers.zip](https://github.com/Jermolene/TiddlyWiki5/files/6948590/tiddlers.zip)

**Comment by kookma** [comment, marked as answer]
Saq
The below code works, but I do not understand the reason

```
\define ff() [tag[xtx]] :sort:$(sortType)$:$(sortNegate)$[split<delimiter>!is[blank]trim[]nth<pos>]

\define m01() $(sortType)$:$(sortNegate)$


\define test2()
<$vars pos=2 sortType=string delimiter=":" sortNegate="reverse"> 

<$list filter=<<ff>>>

</$list>

</$vars>
\end

<$log test2=<<test2>>/>

<<test2>>
```


Note to `<$list filter=<<ff>>>` and my previous filter  `<$list filter="[tag[xtx]] $(ff)$ ">`
